manifests/vhost.pp

```diff
@@ -251,8 +251,12 @@
   # This ensures that the docroot exists
   # But enables it to be specified across multiple vhost resources
   if ! defined(File[$docroot]) and $manage_docroot {
+    $docroot_ensure = $ensure? {
+      present => 'directory',
+      default => 'absent',
+    }
```

Comment:
hmm, we might expect that "ensure => absent" removes all vhost resources, don't we?
I guess (not sure if possible) ```docroot_ensure``` default value set to ```$ensure``` might be for the best. Will give it a try.